src/SessionModel.js

```diff
@@ -108,57 +108,58 @@ export let sessionModel = {
         // recreate local data using the data from Firebase and the API.
 
         const sessionIsValid = await checkValidSessionID(sessionIdFromUI);
-        if(sessionIsValid){
+        if (sessionIsValid) {
             const playerExistsInFB = await checkIfPlayerExists(sessionIdFromUI, this.user.uid); // Boolean
-            if(playerExistsInFB){
+            if (playerExistsInFB) {
                 // The player already exists in the session
                 await this.reJoinSession(sessionIdFromUI, newPlayerName)
-            }else{
+            } else {
                 // Creates a new player which is added to the session
-                if(this.localNumberOfPlayers < 1 || this.localNumberOfPlayers === null){
+                if (this.localNumberOfPlayers < 1 || this.localNumberOfPlayers === null) {
                     this.sessionID = sessionIdFromUI;
                     const player = await this.createPlayer(newPlayerName, this.user.uid, false)
                     await this.dealCards(player.playerID, 5); // always deals five cards
                     playerFBCounter(); // Adds one to the FBCounter
                     this.readyToWriteFB = true;
-                }else{
+                } else {
                     //If one player already has joined on one device.
                     throw new Error("Only one player per device is supported!");
-                }                
+                }
             }
-        }else{
+        } else {
             throw new Error("SessionID is not valid!");
         }
     },
 
-    async reJoinSession(sessionIdFromUI, newPlayerName){
-        if(this.localNumberOfPlayers < 1 || this.localNumberOfPlayers === null){
+    async reJoinSession(sessionIdFromUI, newPlayerName) {
+        if (this.localNumberOfPlayers < 1 || this.localNumberOfPlayers === null) {
             this.sessionID = sessionIdFromUI;
-            console.log("this.sessionID ",  this.sessionID);
+            console.log("this.sessionID ", this.sessionID);
             const isHost = await checkHostFB(this.sessionID, this.user.uid);
             await this.reCreatePlayer(newPlayerName, this.user.uid, isHost);
             this.readyToWriteFB = true;
-        }else{
+        } else {
             throw new Error("Only one player per device is supported!");
-        } 
+        }
     },
 
-    async reJoinSessionURL(sessionIdFromURL, userIDFromURL){
+    async reJoinSessionURL(sessionIdFromURL, userIDFromURL) {
         console.log("reJoinSessionURL")
-        if(this.localNumberOfPlayers < 1 || this.localNumberOfPlayers === null){
+        if (this.localNumberOfPlayers < 1 || this.localNumberOfPlayers === null) {
             this.sessionID = sessionIdFromURL;
+            this.playerOrder = await getPlayerOrderFB(sessionIdFromURL); //*NEW update playerOrder
```

Comment:
This should not be necessary since when the sessionID is set the local data should be updated from FB